apps/api/src/TransformUser/TransformUser.service.ts

```diff
@@ -0,0 +1,75 @@
+import { User, Userfront } from 'types';
+import { InjectRepository } from '@nestjs/typeorm';
+import { Injectable, BadRequestException } from '@nestjs/common';
+import { Repository } from 'typeorm';
+
+@Injectable()
+export class TransformUserService {
```

Comment:
Syntax is quite nice and clean in this service. My only remark would be that you are calling your protected methods only from inside this service, and in each one of them you are making a query to the database to get the same object user (especially for victories and defeats). Wouldn't it be better to give the entity queried from the db as a parameter to avoid multiple calls?

Reply:
en effet c'est mieux mais fleme de tout changer... je vais laisser comme ca pour le moment